apps/example/src/index.ts

```diff
@@ -72,7 +116,14 @@ new Yab()
 		}),
 	)
 	.use(statics("/public", { assetsDir: "./public" }))
-	.use(router("/api", [UserController]))
+	.use(
+		router("/api", [UserController], {
+			validation: async (schema: any, payload: any) => {
+				console.log(2222);
+			},
+			middlewares: [AnyMiddleware],
```

Comment:
we will have some middleware which need to construct with argument. we should store different.

Expect:
```ts
class AuthMiddleware {
  constructor(options?: AuthOptions){}
}

// const Use = <M extends AnyClass, O extends ConstructorParameters<M>[0]>(
//   middleware: M,
//   options?: O
// ) => MethodDecorator;

export const UseAuth(options?: AuthOptions) => Use(AuthMiddleware, options);

```

Reply:
that can be postpone to next PR